plugins/org.eclipse.n4js/src/org/eclipse/n4js/resource/N4JSResource.java

```diff
@@ -259,6 +265,32 @@ public N4JSResource() {
 	}
 
 	/**
+	 * Returns the {@link ASTMetaInfoCache} (in states {@link #isFullyProcessed() "Fully Processed"} and during the
+	 * transition from "Fully Initialized" to "Fully Processed", i.e. during post-processing) or throws an exception if
+	 * the cache is unavailable (in all other states).
+	 */
+	public ASTMetaInfoCache getASTMetaInfoCache() {
+		if (astMetaInfoCache == null) {
+			if (!isFullyProcessed() && !isPostProcessing()) {
+				// getter invoked in wrong state
+				throw new IllegalStateException(
+						"AST meta-info cache only available in state 'Fully Processed' and during post-processing");
+			} else {
+				// getter invoked in correct state, but cache is still undefined
+				throw new NullPointerException("AST meta-info cache missing");
+			}
+		}
+		return astMetaInfoCache;
+	}
+
+	/**
+	 * Set the receiving resource's {@link ASTMetaInfoCache}. Should only be called at the beginning of post-processing.
```

Comment:
The comment is misleading, this method is also called when clearing the cache.

Reply:
done